cli/src/commands/describe.rs

```diff
@@ -218,13 +242,21 @@ pub(crate) fn cmd_describe(
             .collect_vec(),
         |rewriter| {
             let old_commit_id = rewriter.old_commit().id().clone();
+            let new_author = author_arg.clone().map(|(name, email)| Signature {
+                name,
+                email,
+                timestamp: rewriter.old_commit().author().timestamp.clone(),
```

Comment:
https://github.com/martinvonz/jj/blob/56dbbb8fc629a80173b1253e6baec225bc83b8f6/lib/src/commit_builder.rs#L208 modifies the timestamp, so I'm reading it before `rewritter.rebase` is called.

Reply:
I think the updated timestamp should be preserved in that case. For consistency, `jj describe` should update author timestamp if the commit was empty.

Maybe adjust the doc string instead?

Reply:
That would be inconsistent with Git and also lead to strange-looking history that has nonchronological timestamps.

Reply:
`jj` always had that, see #3906 and it's associated bug (#2000). There's no reason that we have to be consistent with Git here.

Reply:
I fully agree with regard to resetting the date of `jj new` commit, not so much about resetting timestamp of other commits but updated PR to that version.
Since this option is now the same as `JJ_USER='Foo Bar' JJ_EMAIL=foo@bar.com jj describe --reset-author` I'm not sure if  there is much benefit from it, shorter and more intuitive syntax seems like the only selling point.

Reply:
I think it was intended as that, so to my understanding this is also fine.

Reply:
> Since this option is now the same as `JJ_USER='Foo Bar' JJ_EMAIL=foo@bar.com jj describe --reset-author`

`--reset-author` always updates author timestamp, but `--author` doesn't, right? I don't think Git consistency is important here, but it's somewhat consistent because in Git, we don't (usually) have a commit without description and content.

Reply:
Sorry, my bad.

I had based this entirely on my misunderstanding of the unit test. Commits there have timestamps rewritten because they have empty descriptions, right?

Running on the real repository shows the author timestamp intact and committer timestamp updated as intended:
```
/tmp/author-test
❯ ~/Projects/jj/target/debug/jj log -T "description ++ ' ' ++ author.timestamp() ++ ' ' ++ committer.timestamp()"
@   2024-09-17 01:41:42.000 +02:00 2024-09-17 01:41:42.000 +02:00
○  bar
│   2024-09-17 01:41:42.000 +02:00 2024-09-17 01:41:42.000 +02:00
○  foo
│   2024-09-17 01:41:32.000 +02:00 2024-09-17 01:41:33.000 +02:00
◆   1970-01-01 00:00:00.000 +00:00 1970-01-01 00:00:00.000 +00:00

/tmp/author-test
❯ jj log
@  vmrplqvo mati865@gmail.com 2024-09-17 01:41:42 4da73e4d
│  (empty) (no description set)
○  ztxlzwmq mati865@gmail.com 2024-09-17 01:41:42 2919a8bd
│  bar
○  xonnomqr mati865@gmail.com 2024-09-17 01:41:33 6359b7c7
│  foo
◆  zzzzzzzz root() 00000000

/tmp/author-test
❯ ~/Projects/jj/target/debug/jj desc @-- --no-edit --author "foo bar <foo@example.com>"
Rebased 2 descendant commits
Working copy now at: vmrplqvo 7bacbeed (empty) (no description set)
Parent commit      : ztxlzwmq 9c1a7191 bar

/tmp/author-test
❯ ~/Projects/jj/target/debug/jj log -T "description ++ ' ' ++ author.timestamp() ++ ' ' ++ committer.timestamp()"
@   2024-09-17 01:47:25.000 +02:00 2024-09-17 01:47:25.000 +02:00
○  bar
│   2024-09-17 01:41:42.000 +02:00 2024-09-17 01:47:25.000 +02:00
○  foo
│   2024-09-17 01:41:32.000 +02:00 2024-09-17 01:47:25.000 +02:00
◆   1970-01-01 00:00:00.000 +00:00 1970-01-01 00:00:00.000 +00:00
```

I have updated the docstrings to indicate timestamp preservation if the description is nonempty.